Assets/Mirror/Editor/MirrorWelcomeWindow.cs

```diff
@@ -0,0 +1,413 @@
+        firstStartUpKey = GetStartUpKey();
+        if(EditorPrefs.GetBool(firstStartUpKey, false) == false && firstStartUpKey != "MirrorUnknown")
+        {
+            OpenWindow();
+            //now that we have seen the welcome window, set this this to true so we don't load the window every time we recompile (for the current version)
+            EditorPrefs.SetBool(firstStartUpKey, true);
+        }
+
+        EditorApplication.update -= ShowWindowOnFirstStart;
+    }
+
+    //open the window (also openable through the path below)
+    [MenuItem("Mirror/Welcome")]
+    public static void OpenWindow()
+    {
+        mirrorIconPath = GetMirrorIconPath();
+
+        //create the window
+        MirrorWelcomeWindow window = GetWindow<MirrorWelcomeWindow>("Mirror Welcome Page");
+        //set the position and size
+        window.position = new Rect(new Vector2(100, 100), windowSize);
+        //set min and max sizes so we cant readjust window size
+        window.maxSize = windowSize;
+        window.minSize = windowSize;      
+    }
+
+    #endregion
+
+    //display the welcome menu
+    private void OnGUI()
+    {
+        #region Styles
+
+        //styles for the various elements on the window (named accordingly)
+
+        GUIStyle versionTextStyle = new GUIStyle
+        {
+            normal = new GUIStyleState
+            {
+                textColor = Color.white
+            },
+            fontSize = 10,
+            padding = new RectOffset(0, 5, 5, 0),
+            alignment = TextAnchor.UpperRight,
+        };
+
+        GUIStyle bannerStyle = new GUIStyle
+        {
+            normal = new GUIStyleState
+            {
+                background = iconBackground,
+            },
+        };
+
+        GUIStyle iconStyle = new GUIStyle
+        {
+            fixedHeight = 64,
+            fixedWidth = 64,
+            alignment = TextAnchor.MiddleCenter,
+        };
+
+        GUIStyle titleStyle = new GUIStyle 
+        { 
+            normal = new GUIStyleState 
+            { 
+                textColor = Color.white 
+            }, 
+            fontSize = 30, 
+            alignment = TextAnchor.MiddleCenter,
+        };
+
+        GUIStyle descriptionStyle = new GUIStyle
+        {
+            normal = new GUIStyleState
+            {
+                textColor = Color.white
+            },
+            fontSize = 14,
+            padding = new RectOffset(5, 5, 5, 5),
+            wordWrap = true,
+        };
+
+        GUIStyle headerStyle = new GUIStyle
+        {
+            normal = new GUIStyleState
+            {
+                textColor = Color.white
+            },
+            fontStyle = FontStyle.Bold,
+            fontSize = 20,
+            padding = new RectOffset(5, 5, 5, 10),
+            wordWrap = true,
+        };
+
+        #endregion
+
+        #region Draw the banner and logo
+
+        //create banner
+        GUILayout.BeginArea(new Rect(new Vector2(0, 0), bannerSize), bannerStyle);
```

Comment:
Might be better to use GUILayout.AreaScope instead of Begin/End.

Shoudl be like this 
```cs
using (new GUILayout.AreaScope(new Rect(new Vector2(0, 0), bannerSize), GUIContent.none, bannerStyle))
{
    GUI.Label(new Rect(iconPosition, iconSize), mirrorIcon, iconStyle);
}
```